### PR TITLE
Make logger message optional in `filter_nonfinite`

### PR DESCRIPTION
**Pull request by xadrianzetx**
## Motivation

This https://github.com/optuna/optuna/pull/3435#issuecomment-1086800869.

## Description of the changes

* Add argument to `filter_nonfinite` making logger messages optional
* Test

### CODE REVIEW

**Comment by xadrianzetx** [comment]
CC @nzw0301 I think we could go with logger lvl `Warning` here if necessary.

**Comment by nzw0301** [comment]
Brilliant! Honestly, I'm not sure which level is appropriate. But I suppose `_logger.warning` is better than `_logger.info` because the objective or returned values by `target` should not take `inf/-inf`.  

@HideakiImamura What do you think about this because you were one of reviewers of https://github.com/optuna/optuna/pull/3395?

**Comment by HideakiImamura** [comment]
I will review this PR in this week.

@nzw0301 Could you review this PR?

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. As @nzw0301 suggested, I agree with using `_logger.warning` instead of `_logger.info` here.

**Comment by nzw0301** [comment]
> @nzw0301 Could you review this PR?

@HideakiImamura Sure, I'll review this PR. Thank you for your comment!

**Comment by xadrianzetx** [comment]
Thanks for the opinion @nzw0301, @HideakiImamura! PTAL at 8fcf378c10d2bd4cea982ff59e9678835361a89d.

File: tests/visualization_tests/test_utils.py [hunk 1]
```diff
@@ -137,3 +140,35 @@ def _target(t: FrozenTrial) -> float:
...
+
+    optuna.logging.enable_propagation()
+    _filter_nonfinite(study.get_trials(states=(TrialState.COMPLETE,)), with_message=with_message)
+    msg = "Trial 1 is omitted in visualization because its objective value is inf or nan."
+
+    if with_message:
+        assert msg in caplog.text
+        for record in caplog.records:
+            if record.msg == msg:
+                assert record.levelno == logging.WARNING
```
Comment (HideakiImamura) [inline tests/visualization_tests/test_utils.py:172]: Why not ensure that this part of the code is executed exactly once?
```suggestion
        n_filtered_as_inf = 0
        for record in caplog.records:
            if record.msg == msg:
                assert record.levelno == logging.WARNING
                n_filtered_as_inf += 1
        assert n_filtered_as_inf == 1
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the update. I have a suggestion for the test. PTAL.

**Review by nzw0301 — APPROVED** [review]
LGTM. Thanks!

**Review by HideakiImamura — APPROVED** [review]
Thanks for the quick fix! LGTM!

**Pull request merged**